File: sprint3/JS/lesson14code1.js
function countingSort(array, k) {
  const countedValues = new Array(k).fill(0);
  for (const value of array) {
    countedValues[value]++;
  }

  let index = 0;
  for (let value = 0; value < k; value++) {
    for (let amount = 0; amount < countedValues[value]; amount++) {
      array[index] = value;
      index++;
    }
  }
  return array;
}

File: sprint3/go/lesson14code1.go
func countingSort(array []int, k int) []int {
	countedValues := make([]int, k)
	for _, value := range array {
		countedValues[value]++
	}

	index := 0
	for value := 0; value < k; value++ {
		for amount := 0; amount < countedValues[value]; amount++ {
			array[index] = value
			index++
		}
	}
	return array
}
